dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/dual/HibernateBundle.java

```diff
@@ -0,0 +1,82 @@
+package io.dropwizard.hibernate.dual;
+
+import static java.util.Objects.requireNonNull;
+
+import io.dropwizard.db.PooledDataSourceFactory;
+import io.dropwizard.hibernate.SessionFactoryFactory;
+import io.dropwizard.hibernate.SessionFactoryHealthCheck;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.util.Duration;
+import org.hibernate.SessionFactory;
+
+import java.util.List;
+
+public abstract class HibernateBundle<T> extends io.dropwizard.hibernate.HibernateBundle<T>
+{
+    public static final String PRIMARY = "hibernate-primary";
+    public static final String READER = "hibernate-reader";
+
+    protected HibernateBundle(Class<?> entity, Class<?>... entities) {
+        super(entity, entities);
+    }
+
+    protected HibernateBundle(List<Class<?>> entities,
+                              SessionFactoryFactory sessionFactoryFactory) {
+        super(entities, sessionFactoryFactory);
+    }
+
+    @Override
+    protected String name() {
+        return PRIMARY;
+    }
+
+    protected String reader() {
+        return READER;
+    }
+
+    abstract public PooledDataSourceFactory getReadSourceFactory(T configuration);
+
+    @Override
+    public final void run(T configuration, Environment environment) throws Exception {
+        final PooledDataSourceFactory primaryConfig = getDataSourceFactory(configuration);
+        final SessionFactory primary = requireNonNull(sessionFactoryFactory.build(this, environment, primaryConfig,
+            entities, name()));
+        final PooledDataSourceFactory readerConfig = getDataSourceFactory(configuration);
```

Comment:
Shouldn't this call `getReadSourceFactory(configuration)`?

Reply:
Yes, thank you for the catch. It's now fixed on the feat/3542 branch.